Harvest.Web/ClientApp/src/Tickets/TicketAttachments.tsx

```diff
@@ -1,28 +1,82 @@
-﻿import { useMemo } from "react";
-import { TicketAttachment } from "../types";
+﻿import { useMemo, useState } from "react";
+import { TicketAttachment, TicketDetails, BlobFile } from "../types";
+import { Button, FormGroup, Input, Label } from "reactstrap";
+import { FileUpload } from "../Shared/FileUpload";
 
 interface Props {
+  ticket: TicketDetails;
   attachments: TicketAttachment[];
+  projectId: string;
+  setTicket: (ticket: TicketDetails) => void;
 }
 
 export const TicketAttachments = (props: Props) => {
+  const { ticket, setTicket, projectId } = props;
   const ticketAttachments = useMemo(() => props.attachments, [
     props.attachments,
   ]);
 
-  if (ticketAttachments === undefined || ticketAttachments.length === 0) {
-    return (
-      <div>
-        <h2>Ticket Attachments</h2>
-        <p>No Attachments</p>
-      </div>
-    );
-  }
+  const [ticketLoc, setTicketLoc] = useState<TicketDetails>({
+  } as TicketDetails);
+
+    const updateFiles = async (attachments: BlobFile[]) => {
+        const response = await fetch(`/Ticket/UploadFiles/${projectId}/${props.ticket.id}/`, {
+          method: "POST",
+          headers: {
+              Accept: "application/json",
+              "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ Attachments: attachments }),
+        });
+
+        if (response.ok) {
+            const data = await response.json() as TicketAttachment[];
+
+            setTicket({ ...ticket, attachments: [...ticket.attachments, ...data] });
```

Comment:
personally I would rather have you keep `setTicket` as `props.setTicket` just so it's clear in this case since it's competing with a local setTicketLoc which is so similar.  Also really we probably need to find a better way to track this new files thing, having a shadow local state var is not great.

Reply:
Yeah, agree. But I was having problems getting it to work.